src/v/iceberg/manifest_list_avro.cc

```diff
@@ -0,0 +1,203 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/manifest_list_avro.h"
+
+#include "base/units.h"
+#include "bytes/iobuf.h"
+#include "iceberg/avro_utils.h"
+#include "iceberg/manifest_file.avrogen.h"
+#include "strings/string_switch.h"
+
+#include <avro/DataFile.hh>
+
+#include <stdexcept>
+#include <vector>
+
+namespace iceberg {
+
+namespace {
+
+constexpr int32_t content_type_to_int(manifest_file_content t) {
+    switch (t) {
+    case manifest_file_content::data:
+        return 1;
+    case manifest_file_content::deletes:
+        return 2;
+    }
+}
+
+manifest_file_content content_type_from_int(int t) {
+    if (t == 1) {
+        return manifest_file_content::data;
+    }
+    if (t == 2) {
+        return manifest_file_content::deletes;
+    }
```

Comment:
as long as all of these values are constrained/checked, i'm pretty sure it's legal to static cast from integer to enum class. are these values coming from external sources or something?

Reply:
Yeah, these will be coming from S3 files, so I don't think we can safely assume this integer is in enum range.

Reply:
Done

Reply:
makes sense, but it's a bit of a footgun if there are ever gaps in the enum range, then checking min/max doesn't work anymore. maybe just guard against that with a note on the enum that the range needs to be contiguous otherwise the conversion routine needs to be changed.

Reply:
Seems unlikely.

Done